src/core/namespace.c

```diff
@@ -1048,11 +1091,9 @@ int setup_namespace(
                 if (r < 0)
                         goto finish;
 
-                if (namespace_info_mount_apivfs(root_directory, ns_info)) {
-                        r = append_static_mounts(&m, apivfs_table, ELEMENTSOF(apivfs_table), ns_info->ignore_protect_paths);
-                        if (r < 0)
-                                goto finish;
-                }
+                r = append_mount_apivfs(&m, root_directory, ns_info, ns_info->ignore_protect_paths);
+                if (r < 0)
+                        goto finish;
```

Comment:
@poettering OK I switched to case where `MountAPIVFS=yes` is implied if ProtectControlGroups= and ProtectKernelTunables= are set which is of course simple. However in this case instead of having a bind mount of /dev from outside, it automatically switches to a PrivateDevices= like mount. So `ProtectControlGroups=` and `ProtectKernelTunables=` now imply `MountAPIVFS=yes` and `PrivateDevices=yes` . This was also documented. If you don't agree feel free to ignore this patch I made it the last one.

Thanks!